add build offset past earlier libgfortran 5.0 outputs

minrk opened the pull request:
since this publishes libgfortran 5.0 which also has past builds (up to [build number 32](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Fosx-arm64%2Flibgfortran-5.0.0-12_2_0_hd922786_32.conda)), some installs may prioritize a higher (but older) build of libgfortran over a higher version of libgfortran5.

If you request `libgfortran` directly, e.g. `mamba create -p /tmp/env --dry-run libgfortran`, you get outdated builds:

```
  + libgfortran     5.0.0  12_2_0_hd922786_32  conda-forge      162kB
  + libgfortran5   12.2.0  h0eea778_32         conda-forge        1MB
  + llvm-openmp    20.1.2  hdb05f8b_0          conda-forge     Cached
```

`pixi` seems to resolve slightly differently than mamba, where e.g. `numpy` will resolve to libgfortran5 13 (`-` a pixi env with `numpy`, `+` is the same env with constraint `libgfortran5=14.*`):

```
-libgfortran      5.0.0      13_2_0_hd922786_3     107.6 KiB  conda  https://conda.anaconda.org/conda-forge/
-libgfortran5     13.2.0     hf226fd6_3            974 KiB    conda  https://conda.anaconda.org/conda-forge/
+libgfortran      5.0.0      14_2_0_h6c33f7e_2     152.2 KiB  conda  https://conda.anaconda.org/conda-forge/
+libgfortran5     14.2.0     h6c33f7e_2            787.3 KiB  conda  https://conda.anaconda.org/conda-forge/
```

I _believe_ setting the build offset here so that latest builds of `libgfortran=5.0.0` actually have the highest build number should solve that ambiguity.

minrk committed:
add build offset past earlier libgfortran 5.0 outputs

since this publishes libgfortran 5.0 which also has past builds (up to build number 32),
some installs may prioritize a higher build of libgfortran over a higher version of libgfortran5
so make sure this feedstock's outputs win on both fronts

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-forge-tick (the bot)`. Your recipe  may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.
 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14267222920](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14267222920). Examine the logs at this URL for more detail.</sub>

beckermr commented on recipe/meta.yaml:
```diff
@@ -1,5 +1,5 @@
 {% set chost = macos_machine %}
-{% set build_number = 2 %}
+{% set build_number = 102 %}
```
This looks good but someone might reset to zero. Can we embed the offset inside the build number in the recipe as a jinja2 expression with `{{ build_number | int + 100 }}`?

minrk replied:
Good idea. I did it with a `build_offset` variable and a comment that resetting to 0 is not likely to be safe in this feedstock which publishes multiple versions (resetting to 0 will only ever be okay if _all_ output versions are increased to previously unpublished values in the same PR)

minrk replied:
We could also apply the offset only to the duplicate-version outputs, which I _think_ is `libgfortran` and `libgfortran-devel`, but I can check if that would be preferable.

beckermr replied:
I'd rather have all of the build numbers be the same to help keep track of when things are built. So the current code looks good.

minrk committed:
track build_offset separately to avoid accidentally reseting back to 0

and add a note that resetting to 0 is only safe if no versions will be republished

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

I considered the following status checks when analyzing this PR:
 - **linter**: passed
 - **azure**: passed


Thus the PR was passing and merged! Have a great day!
